libkbfs/folder_block_ops.go

```diff
@@ -689,8 +699,8 @@ func (fbo *folderBlockOps) cacheBlockIfNotYetDirtyLocked(
 	lState *lockState, ptr BlockPointer, branch BranchName, block Block) error {
 	fbo.blockLock.AssertLocked(lState)
 
-	if !fbo.config.BlockCache().IsDirty(ptr, branch) {
-		return fbo.config.BlockCache().PutDirty(ptr, branch, block)
+	if !fbo.config.DirtyBlockCache().IsDirty(ptr, branch) {
+		return fbo.config.DirtyBlockCache().Put(ptr, branch, block)
```

Comment:
No, nothing should get removed from the clean block cache except (a) by LRU, and (b) when all references to a block are removed and it can no longer be de-dupped.  Clean blocks are immutable and so can stick around as long as we have space and can still be referenced.  We are effectively "branching" the block here, and it will eventually get a new block ID when it becomes clean.